packages/gatsby/src/commands/develop.ts

```diff
@@ -349,7 +353,17 @@ async function startServer(program: IProgram): Promise<IServer> {
   return { compiler, listener, webpackActivity }
 }
 
+const feedbackHandler = async (exitCode: number | string): Promise<void> => {
+  if (await userPassesFeedbackRequestHeuristic()) {
+    showFeedbackRequest()
+  }
+  process.exit(exitCode === 1 ? 1 : 0)
+}
+
 module.exports = async (program: IProgram): Promise<void> => {
+  process.on(`exit`, feedbackHandler)
```

Comment:
`exit` event won't work with async function:
> Listener functions must only perform synchronous operations.
 https://nodejs.org/api/process.html#process_event_exit )

Check `beforeExit` event - https://nodejs.org/api/process.html#process_event_beforeexit

Reply:
Additionally we do use 'signal-exit' wrapper package in couple of places which handle exiting in lot of scenarios. I don't think it handles async callback, but worth a try (or checking out code if that package)

Reply:
ooo good point. I think I probably don't need the exit event now that I think about it. I really should only show the feedback if a user intentionally quits. an exit event can be caused by several reasons where the user wouldn't even be looking at the terminal.